packages/services/src/setting/index.ts

```diff
@@ -50,6 +55,10 @@ export class SettingManager extends DataConnector<
       throw new Error('Plugin `id` parameter is required for settings fetch.');
     }
 
+    if (Private.isDisabled(id)) {
+      throw new Error(`Plugin ${id} is disabled.`);
```

Comment:
Shouldn't this and the `id` check come before the response creation + await? If that order is significant, please add a comment explaining why 😄

Reply:
You are correct. This probably happened after we switched to async/await and is a mistake.